controllers/userQuizQuestion.js

```diff
@@ -125,11 +126,47 @@ const getUserQuizQuestionOptionsByQuestion = async (question) => {
     return packOptions(answers)
 }
 
+const getUserAnswerIDs = async (userQuiz) => {
+    const userAnswers = await getUserQuizQuestions(userQuiz)
+
+    const userAnswerIDsRequest = userAnswers.map(async (userAnswer) => {
+        let userAnswerObj = userAnswer.userAnswerObj
+        return await userAnswerObj.answer.get()
+    });
+
+    let userAnswerIDs = []
+    
+    Promise.all(userAnswerIDsRequest).then((answerIDObjs) => {
+        answerIDObjs.forEach((obj) => {
+            userAnswerIDs.push(obj.id)
+        })
+    })
+
+    return userAnswerIDs
+}
+
+const submitUserQuizQuestions = async (userQuizID, userAnswers, correctAnswers) => {
+    // calculate score
+    let score = 0
+    userAnswers.forEach((answer, index) => {
+        if (answer === correctAnswers[index]) {
+            score += 1
+        }
+    });
```

Comment:
try practice using `Array.reduce()`.

This is a very good case to use reduce, highly recommend u try it out cause u don't get many chances to use reduce.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce